.github/workflows/doc-linkcheck.yml

```diff
@@ -0,0 +1,56 @@
+name: doc-linkcheck
+run-name: Documentation Link Check
+
+on:
+  pull_request:
+    types: [labeled]
+
+permissions:
+  contents: read
+
+env:
+  HATCH_ENV_TYPE_VIRTUAL_PATH: ${{ github.workspace }}/.hatch_envs
+
+jobs:
+  linkcheck-docs:
+    name: Sphinx Documentation Link Check
+    runs-on: ubuntu-latest
+    # Run only when the specific label is added
+    if: github.event.label.name == 'status | ready for merge'
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          persist-credentials: false
+
+      - uses: ./.github/actions/setup-docs-env
+
+      - name: Check GitHub API Rate Limit
+        id: rate_limit_check
+        env:
+          GH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        run: |
+          API_URL="https://api.github.com/rate_limit"
+          RESPONSE=$(curl -s -H "Authorization: token $GH_TOKEN" -H "Accept: application/vnd.github.v3+json" $API_URL)
+          REMAINING_CORE=$(echo "$RESPONSE" | jq -r '.resources.core.remaining')
+          echo "Remaining core API calls: $REMAINING_CORE"
+          THRESHOLD=100
```

Comment:
```suggestion
          THRESHOLD=100
          echo "remaining_core=$REMAINING_CORE" >> $GITHUB_OUTPUT
          echo "threshold=$THRESHOLD" >> $GITHUB_OUTPUT
```

Reply:
This would align with https://github.com/teemtee/tmt/pull/3725/files#r2092814080